framework/aster-frame/src/vm/frame.rs

```diff
@@ -728,12 +728,15 @@ impl<'a> VmWriter<'a> {
     /// The size of the available space must be a multiple of the size of `value`.
     /// Otherwise, the method would panic.
     pub fn fill<T: Pod>(&mut self, value: T) {
-        assert!(self.avail() / value.as_bytes().len() > 0);
         assert!(self.avail() % value.as_bytes().len() == 0);
-
-        while self.avail() > 0 {
-            self.write(&mut value.as_bytes().into());
-        }
+        let len = self.avail() / value.as_bytes().len();
+        assert!(len > 0);
+        // Safety: `len` is calculated by the `avail` and the size of the type `T`,
+        // hence this slice is valid and the memory of the slice is managed by this writer.
+        let slice = unsafe { core::slice::from_raw_parts_mut(self.cursor as *mut T, len) };
```

Comment:
First of all, it is unsafe to call `core::slice::from_raw_parts_mut` without checking the alignment of the pointer.

More fundamentally, we should avoid using creating any references (`&T`, `&mut T`, `&[T]`, or `&mut [T]`) out of untyped memory. Untyped memory may be mapped to the user space or DMA mapped for access by devices. So they could be _mutated_ even when we are holding references to them. To access or manipulate untyped memory, use raw pointers and its unsafe methods under the `ptr` module directly.